source/system-administrators/upgrade/upgrading-in-context-editing.rst

```diff
@@ -0,0 +1,221 @@
+- ``data-studio-embedded-item-id``
+
+The new attributes you would be setting are
+
+- ``data-craftercms-model-id``: the UUID of the model you're rendering (i.e. the value of the ``objectId`` tag on the xml)
+- ``data-craftercms-model-path``: path to the model you're rendering
+- ``data-craftercms-field-id``: the id of the field within the content model (only applicable for fields, not for models)
+- ``data-craftercms-index``: the index within the collection (only applicable for collection items, or fields nested inside the repeat group)
+
+As mentioned before, the recommended approach is to use the JavaScript SDK, specifically the
+`@craftercms/experience-builder <https://www.npmjs.com/package/@craftercms/experience-builder>`_ package.
+Once you install or download the package, from its `index`, you should use :ref:`getICEAttributes` which will
+create and return an object with all the necessary attributes and values for you to simply apply the
+attributes to your elements in whatever way suits your app. If you're using React, you should use the
+:ref:`React-specific bindings <xb-react-bindings>` that the package offers.
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Migrating from ``@craftercms/ice``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The ``@craftercms/ice`` package is superseded by `@craftercms/experience-builder <https://www.npmjs.com/package/@craftercms/experience-builder>`_.
+You should replace the usage of the old package with the new.
+
+- The new package also exports function :ref:`getICEAttributes` to get the attributes you must set
+  onto your elements. Their arguments are fairly similar but not identical so be sure to update the
+  arguments too.
+
+- If you were using ``repaintPencils``, you can safely remove it. No replacement is required.
+
+- The use of ``getDropZoneAttributes`` is also replaced with getICEAttributes, no special function is
+  needed with the new function. All fields would use getICEAttributes.
+
+- The ``reportNavigation`` method is `replaced` by :ref:`initExperienceBuilder <js-app-initExperienceBuilder>`.
+  If you are using report navigation, there's a good chance you should initialize XB manually; for that purpose,
+  you should add ``initializeInContextEditing=false`` to your ``crafter.body_bottom`` invocation (i.e.
+  ``<@crafter.body_bottom initializeInContextEditing=false />``). Once you've done that, you're responsible
+  for initializing XB programmatically on your own using ``initExperienceBuilder``. You should invoke,
+  ``initExperienceBuilder`` after each new page is rendered and you want to tell Studio it is now viewing
+  a different page. Before initializing a new view, once the navigation has occurred you should invoke
+  the unmount function that's returned by ``initExperienceBuilder`` (e.g. ``initExperienceBuilder({ ... }).unmount()``).
+
+- The new package also exports ``fetchIsAuthoring`` and ``addAuthoringSupport`` and the methods on both packages are equivalent.
+
+----
```

Comment:
Remove this section

Reply:
I don't think this is necessary. As part of the upgrade activities devs will be removing one package and adding a new one. These methods coming from the old package may create uncertainty as to what to do with these. Don't see any harm in leaving it and it's reassurance on what the upgrade process looks like.